backend/utils/threads_utils.py

```diff
@@ -56,8 +56,40 @@ async def add_message_to_thread_list(org_id, room_id, message_id, request: Threa
     thread_message["created_at"] = str(datetime.utcnow())
 
     message["threads"].insert(0, thread_message)
-   
+
     response = await update_message(org_id, message_id, message)
-    
+
     return response, thread_message
 
+
+async def update_thread_message(
+    org_id: str, message_id: str, thread_id: str, payload: dict[str, Any], message: dict[str, Any]
+) -> dict[str, Any]:
+    """
+     Updates a thread object in message["threads"] then updates the message document in the database.
+     Args:
+         org_id (str): The organization id where the message is being updated.
+         message_id (str): The id of the message to be edited.
+         thread_id (str): The id of the thread to be edited
+         payload (dict[str, Any]): The new data.
+         message (dict[str, Any]): The message to be edited that has been loaded from the database
+     Returns:
+         dict[str, Any]: The response returned by DataStorage's update method.
+     """
+    threads = message["threads"]  # a list of thread objects
+
+    # to edit a thread object in the list of threads
+    for index in range(len(threads)):
+
```

Comment:
Like you already mentioned in your PR description, this is a potential problem. we can't be looping through 100 threads just to update 1

Reply:
We need to find a better implementation here.
But just a quick suggestion, in this situation it's better to use **enumerate** instead of **range+len**.

https://themeptation.medium.com/why-use-enumerate-instead-of-range-in-your-pythons-loops-d33bfd9c7531